### PR TITLE
Feature/healing crystals

### PR DESCRIPTION
**Pull request by TArchetype**

### CODE REVIEW

**Review by SondreHus — CHANGES_REQUESTED** [review]
Nice visual! It behaves a bit quirky when multiple sources of damage happen mid-animation. Also the material is set every frame when animating, and the loot crystal give 1600 currency.  Made a commit to fix these issues"

**Review by SondreHus — APPROVED** [review]

File: Assets/Scripts/UI/UIManager.cs [hunk 1]
```diff
@@ -8,18 +8,34 @@ public class UIManager : MonoBehaviour
...
+    [Range(0, 5)]
+    public float followBarDelay = 2f; 
+
     private InventoryManager inventory;
     private BaseController baseController;
 
     private float baseMaxHealth;
 
-    public int score;
+    private float actualhealth = 0;
```
Comment (Fueredoriku) [inline Assets/Scripts/UI/UIManager.cs:34]: Should be actualHealth

File: Assets/Scripts/Interactables/Loot.cs [hunk 2]
```diff
@@ -87,13 +91,14 @@ private void DestroyAnimation()
                 / meshRenderers[0].material.GetFloat(rateMaterialProperty)
                 > 1)
             {
-                baseController.crystals++;
                 baseController.RemoveRayVFX(transform, 10f);
 
                 //Add resources to inventory
+                baseController.crystals++;
                 inventory.ResourceAmount += lootValue;
+                baseController.GetComponent<HealthLogic>().Heal(baseHealAmount);
```
[the remark this answers is not in the document]
Comment (SondreHus) [reply Assets/Scripts/Interactables/Loot.cs:99]: Not the responsibility of the component healing, causes more clutter than if it is handled by the base controller, which it is. As for the animation, a clamping has been implemented.

[the remark this answers is not in the document]
Comment (Fueredoriku) [reply Assets/Scripts/Interactables/Loot.cs:99]: A clamp does the job too 👍

**Review by Fueredoriku — APPROVED** [review]
Looks good, ver nice changes! 😎🎉
Read through code and tested.

**Pull request merged**